### PR TITLE
Add Django 4.0a1 to tox.ini

### PR DESCRIPTION
**Pull request by matthiask**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -4,15 +4,16 @@ envlist =
     style
     packaging
     py{36,37}-dj{22,31,32}-sqlite
-    py{38,39}-dj{22,31,32,main}-sqlite
+    py{38,39}-dj{22,31,32,40,main}-sqlite
     py{36,37,38,39}-dj{22,31,32}-{postgresql,mysql}
-    py{310}-dj{32,main}-{sqlite,postgresql,mysql}
+    py{310}-dj{32,40,main}-{sqlite,postgresql,mysql}
```
Comment (felixxm) [inline tox.ini:9]: `py{38,39}-{40,main}-{postgresql,mysql}` is missing?

Comment (matthiask) [reply tox.ini:9]: Ah yes, thanks!

I'm *so* confused by this matrix 😅

Comment (felixxm) [reply tox.ini:10]: True, maybe:
```
    py{36,37}-dj{22,31,32}-{sqlite,postgresql,mysql}
    py{38,39}-dj{22,31,32,40,main}-{sqlite,postgresql,mysql}
    py{310}-dj{32,40,main}-{sqlite,postgresql,mysql}
```
it's more readable to me :thinking:

Comment (felixxm) [reply tox.ini:10]: Django 3.2 doesn't officially support Python 3.10.

Comment (matthiask) [reply tox.ini]: Very helpful, thank you!

**Comment by tim-schilling** [comment]
As a point of clarification, I intended for the `dj-main` case to only be tested with sqlite. The reasoning was that it's used as a canary for upcoming incompatible changes and we don't need to test every permutation for it. I'm fine with the changes made in this PR though.

**Pull request merged**